# HG changeset patch
# User Rik <rik@octave.org>
# Date 1544506899 28800
#      Mon Dec 10 21:41:39 2018 -0800
# Node ID 33e6ab3e14917f3c9bc0accffdea22f6b41775bb
# Parent  704cd93ca74898356ce13fe42e9451242da73d39
mkpp.m: Document how ppval will calculate values for each piecewise section (patch #9095).

* mkpp.m: Add Programming Note to documentation explaining how the lower bound
of the piecewise segment will be subtracted from the value of the polynomial
calculated for that segment.

diff --git a/scripts/polynomial/mkpp.m b/scripts/polynomial/mkpp.m
--- a/scripts/polynomial/mkpp.m
+++ b/scripts/polynomial/mkpp.m
@@ -37,16 +37,20 @@
 ## vector-valued or array-valued polynomial.  In that case the polynomial
 ## order @var{m} is defined by the length of the last dimension of @var{coefs}.
 ## The size of first dimension(s) are given by the scalar or vector @var{d}.
 ## If @var{d} is not given it is set to @code{1}.  In this case
 ## @code{@var{p}(@var{r}, @var{i}, :)} contains the coefficients for the
 ## @var{r}-th polynomial defined on interval @var{i}.  In any case @var{coefs}
 ## is reshaped to a 2-D matrix of size @code{[@var{ni}*prod(@var{d}) @var{m}]}.
 ##
+## Programming Note: @code{ppval} evaluates polynomials at
+## @code{@var{xi} - @var{breaks}(i)}, i.e., it subtracts the lower endpoint of
+## the current interval from @var{xi}.  This must be taken into account when
+## creating piecewise polynomials objects with @code{mkpp}.
 ## @seealso{unmkpp, ppval, spline, pchip, ppder, ppint, ppjumps}
 ## @end deftypefn
 
 function pp = mkpp (x, P, d)
 
   ## check number of arguments
   if (nargin < 2 || nargin > 3)
     print_usage ();
